base/broadcast.jl

```diff
@@ -219,6 +223,12 @@ argtype(bc::Broadcasted) = argtype(typeof(bc))
 _eachindex(t::Tuple{Any}) = t[1]
 _eachindex(t::Tuple) = CartesianIndices(t)
 
+Base.IndexStyle(bc::Broadcasted) = IndexStyle(typeof(bc))
+Base.IndexStyle(::Type{<:Broadcasted{<:Any,<:Tuple{Any}}}) = IndexLinear()
+Base.IndexStyle(::Type{<:Broadcasted{<:Any}}) = IndexCartesian()
+
+Base.LinearIndices(bc::Broadcasted{<:Any,<:Tuple{Any}}) = axes(bc)[1]
```

Comment:
Am right that this usually won't return a `LinearIndices`? Should this rather be `LinearIndices(axes(bc))`?

Reply:
Yeah, that'd probably be better.  This sort of specialization should probably be on `eachindex(::IndexLinear, ...)` or `LinearIndices` itself if it's even needed at all.